packages/lib/modules/pool/actions/recovery-mode/RecoveryMode.tsx

```diff
@@ -0,0 +1,126 @@
+import { ActionFooter } from '@repo/lib/shared/components/modals/ActionModalFooter'
+import { usePoolRedirect } from '../../pool.hooks'
+import { AnimateHeightChange } from '@repo/lib/shared/components/animations/AnimateHeightChange'
+import { MobileStepTracker } from '@repo/lib/modules/transactions/transaction-steps/step-tracker/MobileStepTracker'
+import { zeroAddress } from 'viem'
+import { abbreviateAddress } from '@repo/lib/shared/utils/addresses'
+
+export function RecoveryMode() {
+  const { isDesktop } = useBreakpoints()
+  const { userAddress } = useUserAccount()
+  const { pool, isLoading: isPoolLoading, chain } = usePool()
+  const { redirectToPoolPage } = usePoolRedirect(pool)
+  const enableRecoveryModeStep = useRecoveryModeStep()
+
+  const isLoading = isPoolLoading
+  const transactionSteps = useTransactionSteps([enableRecoveryModeStep], isLoading)
+  const txHash = transactionSteps.lastTransaction?.result?.data?.transactionHash
+
+  const receiptProps = useRecoveryModeChangedReceipt({
+    chain,
+    txHash,
+    userAddress,
+    protocolVersion: pool.protocolVersion as ProtocolVersion,
+    txReceipt: transactionSteps.lastTransaction?.result,
+  })
+
+  const isSuccess = !!txHash && !!receiptProps.data && receiptProps.enabled
+
+  return (
+    <Box maxW="lg" mx="auto" pb="2xl" w="full">
+      {isDesktop && (
+        <DesktopStepTracker
+          chain={pool.chain}
+          extraStyles={{ position: 'relative', top: '115px', left: '600px' }}
+          isTxBatch={false}
+          transactionSteps={transactionSteps}
+        />
+      )}
+
+      <Card width="xl">
+        <CardHeader>
+          <TransactionHeader
+            chain={pool.chain}
+            isReceiptLoading={receiptProps.isLoading}
+            label="Enable recovery mode"
+            txHash={txHash}
+          />
+        </CardHeader>
+
+        <CardBody>
+          <RecoveryModeBody isMobile={!isDesktop} pool={pool} transactionSteps={transactionSteps} />
+        </CardBody>
+
+        <CardFooter>
+          <ActionFooter
+            currentStep={transactionSteps.currentStep}
+            isSuccess={isSuccess}
+            returnAction={redirectToPoolPage}
+            returnLabel="Return to pool"
+          />
+        </CardFooter>
+      </Card>
+    </Box>
+  )
+}
+
+type BodyProps = {
+  isMobile: boolean
+  pool: Pool
+  transactionSteps: TransactionStepsResponse
+}
+
+function RecoveryModeBody({ isMobile, pool, transactionSteps }: BodyProps) {
+  const pausedBy =
+    pool.pauseManager === zeroAddress || !pool.pauseManager
+      ? 'Balancer Governance'
+      : abbreviateAddress(pool.pauseManager)
+
+  return (
+    <AnimateHeightChange spacing="ms">
+      {isMobile && <MobileStepTracker chain={pool.chain} transactionSteps={transactionSteps} />}
+
+      <VStack p="ms">
+        <Text fontWeight="bold" w="full">
+          {`This Balancer v3 pool has been paused by ${pausedBy}`}
+        </Text>
```

Comment:
maybe we should make this a link to block explorer for the address?

otherwise could be perceived as we are trying to hide pause manager identity even tho blockchain public lol

Reply:
sure, why not